src/components/video.tsx

```diff
@@ -0,0 +1,243 @@
+   */
+  src: Asset | string;
+
+  /**
+   * Give a fixed width to the video.
+   */
+  width?: number;
+
+  /**
+   * Give a fixed height to the video.
+   */
+  height?: number;
+
+  /**
+   * Set to false to hide the video controls.
+   */
+  controls?: boolean;
+
+  /**
+   * Set a manual data URL to be used as a placeholder image before the poster image successfully loads.
+   * For imported videos this will be automatically generated.
+   */
+  blurDataURL?: string;
+
+  /**
+   * For best image loading performance the user should provide the sizes attribute.
+   * The width of the image in the webpage. e.g. sizes="800px". Defaults to 100vw.
+   * https://developer.mozilla.org/en-US/docs/Web/HTML/Element/img#sizes
+   */
+  sizes?: string;
+
+  /**
+   * A custom function used to resolve video URLs.
+   */
+  loader?: VideoLoader;
+}
+
+export type VideoLoader = (p: VideoLoaderProps) => Promise<string>;
+
+export interface VideoLoaderProps {
+  src: string;
+  width?: number;
+  height?: number;
+}
+
+const DEV_MODE = process.env.NODE_ENV === 'development';
+const FILES_FOLDER = 'videos/';
+const API_ROUTE = '/api/video';
+
+const toSymlinkPath = (path?: string) => {
+  if (!path?.startsWith(FILES_FOLDER)) return path;
+  return path?.replace(FILES_FOLDER, `_${FILES_FOLDER}`);
+}
+
+const NextVideo = forwardRef<MuxPlayerRefAttributes | null, VideoProps>((props: VideoProps, forwardedRef) => {
+  let {
+    src,
+    width,
+    height,
+    controls = true,
+    loader = defaultLoader
+  } = props;
+  let [asset, setAsset] = useState(src);
+
+  // Required to make Next.js fast refresh when the local JSON file changes.
+  // https://nextjs.org/docs/architecture/fast-refresh#fast-refresh-and-hooks
+  if (typeof src === 'object') asset = src;
+
+  const status = typeof asset === 'object' ? asset.status : undefined;
+
+  let { blurDataURL, ...posterProps } = getPosterProps(props, { asset });
+  let videoProps = getVideoProps(props, { asset });
+
+  async function fetchItems(abortSignal: AbortSignal) {
+    if (typeof asset === 'object') return;
+
+    try {
+      const requestUrl = await loader({ src: asset, width, height });
+      const res = await fetch(requestUrl, { signal: abortSignal });
+      const json = await res.json();
+      if (res.ok) {
+        setAsset(json);
+      } else {
+        let message = `[next-video] The request to ${res.url} failed. `;
+        message += `Did you configure the \`${API_ROUTE}\` route to handle video API requests?\n`;
+        throw new Error(message);
+      }
+    } catch (err) {
+      if (!abortSignal.aborted) {
+        console.error(err)
+      }
+    }
+  }
+
+  const needsPolling = DEV_MODE && (typeof asset === 'string' || status != 'ready');
+  usePolling(fetchItems, needsPolling ? 1000 : null);
+
+  const [playing, setPlaying] = useState(false);
+
+  return (
+    <div className="next-video-container">
+      <style>{
```

Comment:
same question as above, is this `styled-jsx`? If so, does it need the `jsx` attr?